cypress/integration/edit_segment.spec.js

```diff
@@ -0,0 +1,384 @@
+    
+        cy.visit(Cypress.env('login_url'));
+        cy.get("[name=csrfmiddlewaretoken]")
+          .should("exist")
+          .should("have.attr", "value")
+          .as("csrfToken");
+    
+        cy.get("@csrfToken").then((token) => {
+          cy.request({
+            method: "POST",
+            url: Cypress.env("login_url"), 
+            form: true,
+            body: {
+              username: "cypress",
+              password: "1234asdf",
+            },
+            headers: {
+              "X-CSRFTOKEN": token,
+            },
+          });
+        });
+    })
+
+    it("shows original word", () => {
+        
+        cy.visit(Cypress.env('home'));
+
+        cy.get(".table")
+            .should("be.visible");
+
+        cy.get('a[name="word-link"]:first')
+            .then((word) => {
+                const w = word[0].id
+                cy.get('a[name="word-link"]:first')
+                    .click()
+
+                cy.location('pathname')
+                    .should('include', Cypress.env("segment_details_url"));
+
+                cy.get('#segment-table')
+                    .contains(w)
+            })
+        })
+    
+    it("shows all tables", () => {
+        cy.visit(Cypress.env("segment_details_url"));
+
+        cy.get("#segment-table").within(() => {
+            cy.get('th').contains('Transcription')
+            cy.get('th').contains('Translation')
+            cy.get('th').contains('Recordings')
+            cy.get('th').contains('Speaker')
+        })
+
+        cy.get("#suggestions-table").within(() => {
+            cy.get('th').contains('Suggestion')
+            cy.get('th').contains('Translation')
+            cy.get('th').contains('Analysis')
+            cy.get('th').contains('MED')
+            cy.get('th').contains('Options')
+        })
+
+        cy.get("#revision-table").within(() => {
+            cy.get('th').contains('User')
+            cy.get('th').contains('Date')
+            cy.get('th').contains('Transcription')
+            cy.get('th').contains('Translation')
+            cy.get('th').contains('Analysis')
+            cy.get('th').contains('Options')
+        })
+
+        cy.get('#edit')
+                .should('not.be.visible')
+    })
+
+    it("shows all buttons", () => {
+        cy.visit(Cypress.env("segment_details_url"));
+
+        cy.get("#suggestions-table").within(() => {
+            cy.get('input:first')
+                .should('have.value', 'Accept')
+        })
+
+        cy.get("#revision-table").within(() => {
+            cy.get('input:first')
+                .should('have.value', 'Revert')
+        })
+
+    })
+
+    it("should load content when clicking Accept", () => {
+        cy.visit(Cypress.env("segment_details_url"));
+
+        cy.get("#suggestions-table").within(() => {
+            cy.get('td:first')
+                .invoke('text')
+                .as('transcription')
+            cy.get('td:nth-child(2):first')
```

Comment:
This is a brittle selector. I would consider doing that `[cy-data]` hack to avoid breaking the test when the text or markup changes.

(this goes for other `nth-child(n)` selectors too!)